contracts/contracts/RaidenMicroTransferChannels.sol

```diff
@@ -140,11 +159,31 @@ contract RaidenMicroTransferChannels {
     /// the `_deposit` token deposit to this contract, compatibility with ERC20 tokens.
     /// @param _receiver_address The address that receives tokens.
     /// @param _deposit The amount of tokens that the sender escrows.
-    function createChannelERC20(address _receiver_address, uint192 _deposit) external {
+    function createChannel(address _receiver_address, uint192 _deposit) external {
         createChannelPrivate(msg.sender, _receiver_address, _deposit);
 
-        // transferFrom deposit from sender to contract
-        // ! needs prior approval from user
+        // transferFrom deposit from msg.sender to contract
+        // ! needs prior approval from msg.sender
+        require(token.transferFrom(msg.sender, address(this), _deposit));
+    }
+
+    /// @notice Function that allows a trusted contract to create a new channel between `_sender_address` and `_receiver_address` and transfers
+    /// the `_deposit` token deposit to this contract, compatibility with ERC20 tokens.
+    /// @param _sender_address The sender's address in behalf of whom the delegate sends tokens.
+    /// @param _receiver_address The address that receives tokens.
+    /// @param _deposit The amount of tokens that the sender escrows.
+    function createChannelDelegate(
+        address _sender_address,
+        address _receiver_address,
+        uint192 _deposit)
+        external
+    {
+        require(trusted_contracts[msg.sender]);
```

Comment:
Suggestion: Turn this into a modifier. i.e.: `callerIsTrustedContract` and use that instead here and in the other functions where you have this require
```
require(trusted_contracts[msg.sender]);
```